normandy/recipes/forms.py

```diff
@@ -0,0 +1,38 @@
+import json
+
+from django import forms
+from django.core.exceptions import ValidationError
+
+import jsonschema
+
+
+class ActionAdminForm(forms.ModelForm):
+    def clean(self):
+        """
+        Validate that the action being edited isn't currently in use.
+        """
+        super().clean()
+
+        if self.instance.in_use:
+            self.add_error(
+                None,
+                ValidationError('Action is currently in use and cannot be edited.')
+            )
+
+
+class RecipeActionInlineForm(forms.ModelForm):
+    def clean(self):
+        """Validate the arguments against their schema."""
```

Comment:
I think kicking it down the road is a good option for now, but you're right that in the long run we want this to be tied to the model rather than the form. I can't think of the reason I chose to go with a form specifically but I do remember trying to do it with model validation at first.